# Ejercicio 3: Modelo Vectorial y TF-IDF  
**Nombre:** Alexis Bautista  
**Fecha de Entrega:** 29 de abril del 2026

## Objetivo de la práctica

- Comprender el modelo vectorial como base para representar documentos y consultas.
- Calcular la matriz TF-IDF para el corpus `data/01_corpus_turismo_500.txt`
- Calcular la matriz TF-IDF para el corpus `Gutenberg 1000`

### Paso 1: Calcular la matriz TF-IDF para el corpus `data/01_corpus_turismo_500.txt`

En este caso se uso el archivo `data/El arte de amar.txt`

In [1]:
with open("data/El arte de amar.txt", "r", encoding="utf-8") as documento:
    contenido = documento.read()

Primero se creo un indice invertido para conocer el numero de veces que aparece un termino en el documento

In [2]:
# convertir todo en minusculas y separar el texto en una lista de palabras
lista_de_palabras = contenido.lower().split()

nombre_documento = "El arte de amar.txt"
indice_invertido = {}

# construccion indice invertido
for palabra in lista_de_palabras:
    
    if palabra not in indice_invertido:
        # Crear palabra en el diccionario principal
        # Su valor será un sub-diccionario con el nombre del doc y el conteo inicial de 1
        indice_invertido[palabra] = {nombre_documento: 1}
        
    else:
        # Acceder a la palabra, luego a su sub-diccionario del documento actual, y sumamos 1
        indice_invertido[palabra][nombre_documento] += 1

print (indice_invertido)

{'nota': {'El arte de amar.txt': 2}, 'de': {'El arte de amar.txt': 1004}, 'transcripción': {'El arte de amar.txt': 2}, '*': {'El arte de amar.txt': 6}, 'las': {'El arte de amar.txt': 467}, 'cursivas': {'El arte de amar.txt': 1}, 'se': {'El arte de amar.txt': 330}, 'muestran': {'El arte de amar.txt': 2}, 'entre': {'El arte de amar.txt': 28}, '_subrayados_.': {'El arte de amar.txt': 1}, 'los': {'El arte de amar.txt': 531}, 'errores': {'El arte de amar.txt': 2}, 'imprenta': {'El arte de amar.txt': 1}, 'han': {'El arte de amar.txt': 15}, 'sido': {'El arte de amar.txt': 9}, 'corregidos.': {'El arte de amar.txt': 1}, 'la': {'El arte de amar.txt': 834}, 'ortografía': {'El arte de amar.txt': 1}, 'del': {'El arte de amar.txt': 178}, 'texto': {'El arte de amar.txt': 2}, 'original': {'El arte de amar.txt': 4}, 'ha': {'El arte de amar.txt': 45}, 'modernizada': {'El arte de amar.txt': 2}, 'acuerdo': {'El arte de amar.txt': 1}, 'con': {'El arte de amar.txt': 445}, 'normas': {'El arte de amar.txt': 1

Por ejemplo la pabra nota aparece 2 veces en el documento `El arte de amar.txt`

In [3]:
# Representacion en tabla
import pandas as pd

df = pd.DataFrame(list(indice_invertido.items()), columns=['Palabra', 'Archivos'])
df


,Palabra,Archivos
0,nota,{'El arte de amar.txt': 2}
1,de,{'El arte de amar.txt': 1004}
2,transcripción,{'El arte de amar.txt': 2}
3,*,{'El arte de amar.txt': 6}
4,las,{'El arte de amar.txt': 467}
...,...,...
8165,cisnes.,{'El arte de amar.txt': 1}
8166,discípulas,{'El arte de amar.txt': 1}
8167,mías,{'El arte de amar.txt': 1}
8168,escriban,{'El arte de amar.txt': 1}


#### **Calculo de la Frecuencia de Terminos (TF)**

In [4]:
import math 
#diccionario para guardar los valores de TF calculados
tf_documento = {}

# recorrer la clave (palabra) y su valor (el sub-diccionario de documentos)
for palabra, documentos in indice_invertido.items():
    
    # frecuencia  (f) buscando el documento en el sub-diccionario
    f = documentos[nombre_documento] 
    
    # formula TF
    tf = 1 + math.log10(f) 
    
    # Guardar el resultado en nuevo diccionario
    tf_documento[palabra] = tf

print(tf_documento)

{'nota': 1.3010299956639813, 'de': 4.001733712809001, 'transcripción': 1.3010299956639813, '*': 1.7781512503836436, 'las': 3.6693168805661123, 'cursivas': 1.0, 'se': 3.5185139398778875, 'muestran': 1.3010299956639813, 'entre': 2.4471580313422194, '_subrayados_.': 1.0, 'los': 3.725094521081469, 'errores': 1.3010299956639813, 'imprenta': 1.0, 'han': 2.1760912590556813, 'sido': 1.9542425094393248, 'corregidos.': 1.0, 'la': 3.921166050637739, 'ortografía': 1.0, 'del': 3.250420002308894, 'texto': 1.3010299956639813, 'original': 1.6020599913279625, 'ha': 2.653212513775344, 'modernizada': 1.3010299956639813, 'acuerdo': 1.0, 'con': 3.6483600109809315, 'normas': 1.0, 'publicadas': 1.0, 'en': 3.727541257028556, '2010': 1.0, 'por': 3.3283796034387376, 'real': 1.0, 'academia': 1.0, 'española.': 1.0, 'también': 2.5440680443502757, 'nombres': 1.4771212547196624, 'propios,': 1.0, 'tanto': 2.146128035678238, 'lugares': 1.6020599913279625, 'como': 3.0043213737826426, 'personajes.': 1.0, 'notas': 1.3010

In [5]:
# Representacion en tabla
df_tf = pd.DataFrame(list(tf_documento.items()), columns=['Palabra', 'TF'])
df_tf

,Palabra,TF
0,nota,1.301030
1,de,4.001734
2,transcripción,1.301030
3,*,1.778151
4,las,3.669317
...,...,...
8165,cisnes.,1.000000
8166,discípulas,1.000000
8167,mías,1.000000
8168,escriban,1.000000


Por ejemplo la frecuancia del termino 'nota' es 1.301

#### **Calculo de la frecuencia inversa del documento (IDF)**

In [6]:
idf_documento = {}
N = 1 # en esta ocasion se lo define en uno porque el ejemplo se hace en un solo documento

# Recorremos índice invertido
for palabra, documentos in indice_invertido.items():
    
    # Calculo df: ¿En cuántos documentos (claves) aparece la palabra?
    df = len(documentos)
    
    # Calculo IDF 
    idf = math.log10(N / df)
    
    # Guardar el resultado en el diccionario
    idf_documento[palabra] = idf
    
print (idf_documento)

{'nota': 0.0, 'de': 0.0, 'transcripción': 0.0, '*': 0.0, 'las': 0.0, 'cursivas': 0.0, 'se': 0.0, 'muestran': 0.0, 'entre': 0.0, '_subrayados_.': 0.0, 'los': 0.0, 'errores': 0.0, 'imprenta': 0.0, 'han': 0.0, 'sido': 0.0, 'corregidos.': 0.0, 'la': 0.0, 'ortografía': 0.0, 'del': 0.0, 'texto': 0.0, 'original': 0.0, 'ha': 0.0, 'modernizada': 0.0, 'acuerdo': 0.0, 'con': 0.0, 'normas': 0.0, 'publicadas': 0.0, 'en': 0.0, '2010': 0.0, 'por': 0.0, 'real': 0.0, 'academia': 0.0, 'española.': 0.0, 'también': 0.0, 'nombres': 0.0, 'propios,': 0.0, 'tanto': 0.0, 'lugares': 0.0, 'como': 0.0, 'personajes.': 0.0, 'notas': 0.0, 'a': 0.0, 'pie': 0.0, 'página': 0.0, 'renemeradas': 0.0, 'y': 0.0, 'colocadas': 0.0, 'tras': 0.0, 'el': 0.0, 'párrafo': 0.0, 'que': 0.0, 'contiene': 0.0, 'su': 0.0, 'llamada.': 0.0, 'páginas': 0.0, 'blanco': 0.0, 'eliminadas.': 0.0, 'arte': 0.0, 'amar.': 0.0, 'traductor': 0.0, 'ovidio': 0.0, 'sus': 0.0, 'obras': 0.0, 'escrito': 0.0, 'otras': 0.0, 'plumas': 0.0, 'más': 0.0, 'bien': 

In [7]:
# Representacion en tabla
df_idf = pd.DataFrame(list(idf_documento.items()), columns=['Palabra', 'IDF'])
df_idf

,Palabra,IDF
0,nota,0.0
1,de,0.0
2,transcripción,0.0
3,*,0.0
4,las,0.0
...,...,...
8165,cisnes.,0.0
8166,discípulas,0.0
8167,mías,0.0
8168,escriban,0.0


Como se hace el ejercicio en un solo documento $(N/df_t)=1$ y $log(1)=0$. Lo que se interpreta como terminos demasiados comunes (la diferencia se ve con mas documentos)

#### **Matriz TF - IDF**

In [8]:
tfidf_documento = {}

#Podemos iterar sobre tf_documento porque tiene exactamente las mismas palabras que idf_documento
for palabra in tf_documento:
    
    # Extraer los valores numericos de ambos diccionarios para esa palabra en especifico
    tf = tf_documento[palabra]
    idf = idf_documento[palabra]
    
    # calculo TF-IDF
    w = tf * idf
    
    # Guardar en el nuevo diccionario
    tfidf_documento[palabra] = w

print(tfidf_documento)


{'nota': 0.0, 'de': 0.0, 'transcripción': 0.0, '*': 0.0, 'las': 0.0, 'cursivas': 0.0, 'se': 0.0, 'muestran': 0.0, 'entre': 0.0, '_subrayados_.': 0.0, 'los': 0.0, 'errores': 0.0, 'imprenta': 0.0, 'han': 0.0, 'sido': 0.0, 'corregidos.': 0.0, 'la': 0.0, 'ortografía': 0.0, 'del': 0.0, 'texto': 0.0, 'original': 0.0, 'ha': 0.0, 'modernizada': 0.0, 'acuerdo': 0.0, 'con': 0.0, 'normas': 0.0, 'publicadas': 0.0, 'en': 0.0, '2010': 0.0, 'por': 0.0, 'real': 0.0, 'academia': 0.0, 'española.': 0.0, 'también': 0.0, 'nombres': 0.0, 'propios,': 0.0, 'tanto': 0.0, 'lugares': 0.0, 'como': 0.0, 'personajes.': 0.0, 'notas': 0.0, 'a': 0.0, 'pie': 0.0, 'página': 0.0, 'renemeradas': 0.0, 'y': 0.0, 'colocadas': 0.0, 'tras': 0.0, 'el': 0.0, 'párrafo': 0.0, 'que': 0.0, 'contiene': 0.0, 'su': 0.0, 'llamada.': 0.0, 'páginas': 0.0, 'blanco': 0.0, 'eliminadas.': 0.0, 'arte': 0.0, 'amar.': 0.0, 'traductor': 0.0, 'ovidio': 0.0, 'sus': 0.0, 'obras': 0.0, 'escrito': 0.0, 'otras': 0.0, 'plumas': 0.0, 'más': 0.0, 'bien': 

In [9]:
# Representacion en tabla
df_tfidf = pd.DataFrame(list(tfidf_documento.items()), columns=['Palabra', 'TF-IDF'])
df_tfidf

,Palabra,TF-IDF
0,nota,0.0
1,de,0.0
2,transcripción,0.0
3,*,0.0
4,las,0.0
...,...,...
8165,cisnes.,0.0
8166,discípulas,0.0
8167,mías,0.0
8168,escriban,0.0


Como el resultado es 0 se concluye que son terminos demasiados comunes, por el mismo hecho de que solo lo hacemos sobre un documento.

### Paso 2: Construir el corpus `Gutenberg 1000`

El corpus `Gutenberg 1000` es un corpus compuesto por 1000 libros de Gutenberg Project

In [10]:
import os

path = "data"
files = os.listdir(path)
N_archivos = len(files)
print("La carpeta",path, "tiene", N_archivos, "documentos")

La carpeta data tiene 1000 documentos


### Paso 3: Calcular la matriz TF-IDF para el corpus `Gutenberg 1000`

Primero se creo un indice invertido para conocer el numero de veces que aparece un termino en los documentos

In [11]:
# diccionario principal
indice_invertido = {}

# Bucle externo: Recorre cada archivo de la lista
for nombre_documento in files:
    
    ruta_completa = os.path.join(path, nombre_documento)
    
    # Abrir y leer el archivo actual
    with open(ruta_completa, "r", encoding="utf-8") as documento:
        contenido = documento.read()
        
    # convertir a minusculas y separar palabraas
    lista_de_palabras = contenido.lower().split()
    
    # Bucle interno: Recorre las palabras del archivo actual
    for palabra in lista_de_palabras:
        
        # Escenario A: La palabra es completamente nueva en todo nuestro índice
        if palabra not in indice_invertido:
            indice_invertido[palabra] = {nombre_documento: 1}
            
        # Escenario B: La palabra ya existe, pero es la primera vez que aparece en ESTE documento en específico
        elif nombre_documento not in indice_invertido[palabra]:
            indice_invertido[palabra][nombre_documento] = 1
            
        # Escenario C: La palabra existe y ya la habíamos visto en ESTE documento, sumamos 1
        else:
            indice_invertido[palabra][nombre_documento] += 1


In [12]:
print(indice_invertido.get("amor"))

{'7 de julio.txt': 2, 'A First Spanish Reader.txt': 1, 'A Flor De Piel Frases.txt': 1, 'A History of Magic and Experimental Science, Volume 1 (of 2) During the First Thirteen Centuries of .txt': 1, 'A los pies de Venus (los Borgia)  $b novela.txt': 35, 'A vuela pluma colección de artículos literarios y políticos.txt': 49, 'A.M.D.G..txt': 8, 'Abel Sánchez Una Historia de Pasión.txt': 7, 'Actas capitulares desde el 21 hasta el 25 de mayo de 1810 en Buenos Aires.txt': 1, 'Adriana Zumarán (novela).txt': 71, 'Adán y Eva en el paraíso.txt': 40, 'Agua de Nieve (Novela).txt': 63, 'Aguas fuertes.txt': 14, 'Al primer vuelo.txt': 7, 'Algo de todo.txt': 52, 'Alma vasca.txt': 14, 'Amar es vencer.txt': 22, 'Amaury.txt': 111, 'Amistad funesta Novela.txt': 29, 'Amor y llanto  $b colección de leyendas históricas originales.txt': 82, 'Amor y Pedagogía.txt': 43, 'Amores elegías amatorias.txt': 56, 'Amparo (Memorias de un loco).txt': 52, 'Américo Vespucio.txt': 3, 'Andanzas y visiones españolas.txt': 32, 

Lo que se interpreta es que la palabra amor aparece 2 veces en el documento `7 de julio.txt`, etc.

In [13]:
# Representacion en tabla
import pandas as pd

df = pd.DataFrame(list(indice_invertido.items()), columns=['Palabra', 'Archivos'])
df

,Palabra,Archivos
0,produced,{'20 poemas para ser leídos en el tranvía.txt'...
1,from,{'20 poemas para ser leídos en el tranvía.txt'...
2,images,{'20 poemas para ser leídos en el tranvía.txt'...
3,available,{'20 poemas para ser leídos en el tranvía.txt'...
4,at,{'20 poemas para ser leídos en el tranvía.txt'...
...,...,...
2543979,cegama!,{'Zumalacárregui.txt': 1}
2543980,cegama...,{'Zumalacárregui.txt': 1}
2543981,"morimos»,",{'Zumalacárregui.txt': 1}
2543982,encomiéndale,{'Zumalacárregui.txt': 1}


#### **Calculo de la Frecuencia de Terminos (TF)**

In [30]:
import math 
#diccionario para guardar los valores de TF calculados
tf_documento = {}

# Bucle externo: recorre la palabra y su sub-diccionario (que contiene varios documentos)
for palabra, documentos in indice_invertido.items():
    
    #sub-diccionario vacío para guardar los TF de esta palabra en específico
    tf_documento[palabra] = {}
    
    # Bucle interno: recorre cada documento donde apareció esta palabra
    # 'nombre_doc' será el nombre del archivo y 'f' será su conteo bruto
    for nombre_doc, f in documentos.items():
        
        # Formula TF
        tf = 1 + math.log10(f) 
        
        # Resultados en un nuevo diccionario
        tf_documento[palabra][nombre_doc] = tf

In [15]:
print(tf_documento.get("amor"))

{'7 de julio.txt': 1.3010299956639813, 'A First Spanish Reader.txt': 1.0, 'A Flor De Piel Frases.txt': 1.0, 'A History of Magic and Experimental Science, Volume 1 (of 2) During the First Thirteen Centuries of .txt': 1.0, 'A los pies de Venus (los Borgia)  $b novela.txt': 2.5440680443502757, 'A vuela pluma colección de artículos literarios y políticos.txt': 2.690196080028514, 'A.M.D.G..txt': 1.9030899869919435, 'Abel Sánchez Una Historia de Pasión.txt': 1.845098040014257, 'Actas capitulares desde el 21 hasta el 25 de mayo de 1810 en Buenos Aires.txt': 1.0, 'Adriana Zumarán (novela).txt': 2.8512583487190755, 'Adán y Eva en el paraíso.txt': 2.6020599913279625, 'Agua de Nieve (Novela).txt': 2.7993405494535817, 'Aguas fuertes.txt': 2.146128035678238, 'Al primer vuelo.txt': 1.845098040014257, 'Algo de todo.txt': 2.7160033436347994, 'Alma vasca.txt': 2.146128035678238, 'Amar es vencer.txt': 2.342422680822206, 'Amaury.txt': 3.0453229787866576, 'Amistad funesta Novela.txt': 2.462397997898956, '

Por ejemplo la palabra amor tiene una frecuencia de 1.30 en el documento `7 de julio.txt`

In [16]:
# Representacion en tabla
df_tf = pd.DataFrame(list(tf_documento.items()), columns=['Palabra', 'TF'])
df_tf

,Palabra,TF
0,produced,{'20 poemas para ser leídos en el tranvía.txt'...
1,from,{'20 poemas para ser leídos en el tranvía.txt'...
2,images,{'20 poemas para ser leídos en el tranvía.txt'...
3,available,{'20 poemas para ser leídos en el tranvía.txt'...
4,at,{'20 poemas para ser leídos en el tranvía.txt'...
...,...,...
2543979,cegama!,{'Zumalacárregui.txt': 1.0}
2543980,cegama...,{'Zumalacárregui.txt': 1.0}
2543981,"morimos»,",{'Zumalacárregui.txt': 1.0}
2543982,encomiéndale,{'Zumalacárregui.txt': 1.0}


#### **Calculo de la frecuencia inversa del documento (IDF)**

In [ ]:
import math 
idf_documento = {}

for palabra, documentos in indice_invertido.items():
    
    # Calculo df: En cuantos documentos (claves) aparece la palabra?
    df = len(documentos)
    
    # IDF 
    idf = math.log10(N_archivos / df)
    
    # Guardar el resultado 
    idf_documento[palabra] = idf

In [32]:
print("IDF de 'amor':", idf_documento.get("amor"))

IDF de 'amor': 0.08354605145007492


In [33]:
# Representacion en tabla
df_idf = pd.DataFrame(list(idf_documento.items()), columns=['Palabra', 'IDF'])
df_idf

,Palabra,IDF
0,produced,0.756962
1,from,0.588380
2,images,0.838632
3,available,0.847712
4,at,0.514279
...,...,...
2543979,cegama!,3.000000
2543980,cegama...,3.000000
2543981,"morimos»,",3.000000
2543982,encomiéndale,3.000000


#### **Matriz TF - IDF**

In [34]:
tfidf_documento = {}

# Bucle externo: Recorre cada palabra
for palabra in tf_documento:
    
    # Preparamos el sub-diccionario vacío para esta palabra
    tfidf_documento[palabra] = {}
    
    # Extrae el valor numérico del IDF global para esta palabra
    idf = idf_documento[palabra]
    
    # Bucle interno: Recorre los documentos específicos donde aparece esa palabra
    # Aquí 'tf' sí será el valor numérico
    for nombre_doc, tf in tf_documento[palabra].items():
        
        # Calculo TF-IDF
        w = tf * idf
        
        tfidf_documento[palabra][nombre_doc] = w


In [35]:
print("TF-IDF de la palabra 'amor':", tfidf_documento.get("amor"))

TF-IDF de la palabra 'amor': {'7 de julio.txt': 0.10869591895583372, 'A First Spanish Reader.txt': 0.08354605145007492, 'A Flor De Piel Frases.txt': 0.08354605145007492, 'A History of Magic and Experimental Science, Volume 1 (of 2) During the First Thirteen Centuries of .txt': 0.08354605145007492, 'A los pies de Venus (los Borgia)  $b novela.txt': 0.21254683972577962, 'A vuela pluma colección de artículos literarios y políticos.txt': 0.22475526011285207, 'A.M.D.G..txt': 0.15899565396735132, 'Abel Sánchez Una Historia de Pasión.txt': 0.1541506557814635, 'Actas capitulares desde el 21 hasta el 25 de mayo de 1810 en Buenos Aires.txt': 0.08354605145007492, 'Adriana Zumarán (novela).txt': 0.23821137669953954, 'Adán y Eva en el paraíso.txt': 0.21739183791166744, 'Agua de Nieve (Novela).txt': 0.23387384957092994, 'Aguas fuertes.txt': 0.17930052328722232, 'Al primer vuelo.txt': 0.1541506557814635, 'Algo de todo.txt': 0.22691135508588847, 'Alma vasca.txt': 0.17930052328722232, 'Amar es vencer.t

Por ejemplo, la palabra amor tiene un TF-IDF de 0.108 en el documento `7 de julio.txt`

In [37]:
# Representacion en tabla
df_tfidf = pd.DataFrame(list(tfidf_documento.items()), columns=['Palabra', 'TF-IDF'])
df_tfidf

,Palabra,TF-IDF
0,produced,{'20 poemas para ser leídos en el tranvía.txt'...
1,from,{'20 poemas para ser leídos en el tranvía.txt'...
2,images,{'20 poemas para ser leídos en el tranvía.txt'...
3,available,{'20 poemas para ser leídos en el tranvía.txt'...
4,at,{'20 poemas para ser leídos en el tranvía.txt'...
...,...,...
2543979,cegama!,{'Zumalacárregui.txt': 3.0}
2543980,cegama...,{'Zumalacárregui.txt': 3.0}
2543981,"morimos»,",{'Zumalacárregui.txt': 3.0}
2543982,encomiéndale,{'Zumalacárregui.txt': 3.0}


### Paso 4: Programar una función `buscar()` para el corpus `Gutenberg 1000`

In [ ]:
def buscar(consulta, top_n=5):
    """
    Busca los documentos más relevantes para una consulta dada.
    top_n: Cantidad de resultados a mostrar (por defecto 5)
    """
    # Limpieza y tokenizacion de la consulta
    palabras_consulta = consulta.lower().split()
    
    # Diccionario temporal para acumular la puntuación de los documentos candidatos
    puntuaciones = {}
    
    # Recorremos cada palabra a buscar
    for palabra in palabras_consulta:
        
        # Verificamos si la palabra existe en nuestra matriz TF-IDF
        if palabra in tfidf_documento:
            
            # Extraemos el sub-diccionario (los libros donde aparece y su peso)
            documentos_con_la_palabra = tfidf_documento[palabra]
            
            # Sumamos el peso TF-IDF al acumulado de cada documento
            for nombre_doc, peso in documentos_con_la_palabra.items():
                if nombre_doc not in puntuaciones:
                    puntuaciones[nombre_doc] = 0.0
                
                # Acumulamos el peso (esto es una versión simplificada del producto punto vectorial)
                puntuaciones[nombre_doc] += peso
                
    # Si ninguna palabra de la consulta estaba en los libros
    if not puntuaciones:
        print(f"No se encontraron resultados para: '{consulta}'")
        return
        
    # Ordenar los resultados de mayor a menor puntuación (Ranking)
    # sorted() ordena un diccionario y lo convierte en una lista de tuplas: [('libro1.txt', 2.5), ...]
    resultados_ordenados = sorted(puntuaciones.items(), key=lambda item: item[1], reverse=True)
    
    # Mostrar en pantalla el Top N de resultados
    print(f"--- Top {top_n} resultados para: '{consulta}' ---")
    for posicion, (documento, score) in enumerate(resultados_ordenados[:top_n], start=1):
        print(f"{posicion}. {documento} | Similitud: {score:.4f}")

In [40]:
buscar("guerra")

--- Top 5 resultados para: 'guerra' ---
1. Historia de la guerra del Peloponeso (1 de 2).txt | Similitud: 0.5154
2. Historia de la guerra del Peloponeso (2 de 2).txt | Similitud: 0.4791
3. Historia de América desde sus tiempos más remotos hasta nuestros días, tomo II.txt | Similitud: 0.4694
4. Historia de la decadencia de España.txt | Similitud: 0.4594
5. Los cuatro jinetes del apocalipsis.txt | Similitud: 0.4571
